trace/trace_test.go

```diff
@@ -83,35 +83,56 @@ func TestTotalTime(t *testing.T) {
 }
 
 func TestLog(t *testing.T) {
-	test := struct {
+	tests := []struct {
 		name             string
+		msg              string
+		msgArgs          []interface{}
 		expectedMessages []string
 		sampleTrace      *Trace
 	}{
-		name: "Check the log dump with 3 msg",
-		expectedMessages: []string{
-			"msg1", "msg2", "msg3",
+		{
+			name: "Check the log dump with 3 msg",
+			expectedMessages: []string{
+				"msg1", "msg2", "msg3",
+			},
+			sampleTrace: &Trace{
+				name: "Sample Trace",
+				steps: []traceStep{
+					{stepTime: time.Now(), msg: "msg1"},
+					{stepTime: time.Now(), msg: "msg2"},
+					{stepTime: time.Now(), msg: "msg3"},
+				},
+			},
 		},
-		sampleTrace: &Trace{
-			name: "Sample Trace",
-			steps: []traceStep{
-				{time.Now(), "msg1"},
-				{time.Now(), "msg2"},
-				{time.Now(), "msg3"},
+		{
+			name: "Check formatting",
+			expectedMessages: []string{
+				"URL=/api,count=3", "msg1 arg1 2 false", "msg2 x",
```

Comment:
Probably for a follow-up PR, should probably test against the entire buffer string instead of matching strings like this.